firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/datagrams.txt

```diff
@@ -0,0 +1,456 @@
+      THEN("The paylaod returned by the output_datagram's paylaod accessor method is equal to the paylaod from the input_buffer body")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+
+  GIVEN("A Datagram receiver of buffer size 254 bytes and expected sequence number equal to 0, with output_datagram constructed with a non-empty payload buffer")
+  // assertions carried out in GIVEN:
+  // The output_datagram's seq accessor method returns 0
+  // The value returned by the output_datagram's length accessor method is 8
+  // The payload buffer returned by the output_datagram's payload accessor method is '0x51 0xb0 0x6e 0xf7 0x86 0x71 0xcd 0x00'  
+
+    WHEN("A body with an incomplete 2-byte header and an empty payload is given to the receiver")
+      THEN("The transform method reports invalid parse status")
+      THEN("After the transform method is called, the value returned by the output_datagram's seq accessor method remains unchanged")
+      THEN("the value returned by the output_datagram's length accessor method remains unchanged")
+      THEN("the paylaod returned by the output_datagram's paylaod accessor method remains unchanged")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+      THEN("the constructor's payload buffer remains unchanged")
+
+    WHEN("transform is called on a body with a complete 2-byte header, and a non-empty paylaod buffer inconsistent with the length field of the header")
+      THEN("The transform method reports invalid length status")
+      THEN("After the transform method is called, the value returned by the output_datagram's seq accessor method is equal to the sequence field of the input_buffer body's header")
+      THEN("The value returned by the output_datagram's length accessor method is equal to the length field of the input_buffer body's header")
+      THEN("The value returned by the output_datagram's length accessor method is inconsistent with the actual length of the payload buffer from the body")
+      THEN("The paylaod returned by the output_datagram's paylaod accessor method is equal to the paylaod from the input_buffer body")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+
+    WHEN("transform is called on a body with a non-empty paylaod buffer and a complete 2-byte header where the sequence field of the header does not match sequence given in output_datagram's constructor")
```

Comment:
I think what matters is that the sequence field of the header doesn't match the DatagramReceiver's expected sequence number. It should be irrelevant what sequence number is given in output_datagram's constructor, the fact that you initialized the DatagramReceiver to expect seq num 0 and you initialized output_datagram with seq num 0 makes it confusing